apps/token/src/routes/governance/components/proposals-list/proposals-list.spec.tsx

```diff
@@ -0,0 +1,227 @@
+  nextMonth,
+  nextWeek,
+} from '../../test-helpers/mocks';
+import type { Proposals_proposals } from '../../proposals/__generated__/Proposals';
+
+const openProposalClosesNextMonth = generateProposal({
+  id: 'proposal1',
+  state: ProposalState.Open,
+  party: {
+    id: 'zxcv',
+  },
+  terms: {
+    closingDatetime: nextMonth.toString(),
+    enactmentDatetime: nextMonth.toString(),
+  },
+  rationale: {
+    description: 'proposal that closes next month',
+  },
+});
+
+const openProposalClosesNextWeek = generateProposal({
+  id: 'proposal2',
+  state: ProposalState.Open,
+  party: {
+    id: 'bvcx',
+  },
+  terms: {
+    closingDatetime: nextWeek.toString(),
+    enactmentDatetime: nextWeek.toString(),
+  },
+  rationale: {
+    description: 'proposal that closes next week',
+  },
+});
+
+const enactedProposalClosedLastWeek = generateProposal({
+  id: 'proposal3',
+  state: ProposalState.Enacted,
+  terms: {
+    closingDatetime: lastWeek.toString(),
+    enactmentDatetime: lastWeek.toString(),
+  },
+});
+
+const rejectedProposalClosedLastMonth = generateProposal({
+  id: 'proposal4',
+  state: ProposalState.Rejected,
+  terms: {
+    closingDatetime: lastMonth.toString(),
+    enactmentDatetime: lastMonth.toString(),
+  },
+});
+
+const failedProposal = generateProposal({
+  id: 'proposal5',
+  state: ProposalState.Failed,
+});
+
+const renderComponent = (proposals: Proposals_proposals[]) => (
+  <Router>
+    <MockedProvider mocks={[networkParamsQueryMock]}>
+      <AppStateProvider>
+        <VegaWalletContext.Provider value={mockWalletContext}>
+          <ProposalsList proposals={proposals} />
+        </VegaWalletContext.Provider>
+      </AppStateProvider>
+    </MockedProvider>
+  </Router>
+);
+
+it('Culls failed proposals', () => {
+  render(renderComponent([failedProposal]));
+  expect(screen.queryByTestId('open-proposals')).not.toBeInTheDocument();
+  expect(screen.getByTestId('no-open-proposals')).toBeInTheDocument();
+  expect(screen.queryByTestId('closed-proposals')).not.toBeInTheDocument();
+  expect(screen.getByTestId('no-closed-proposals')).toBeInTheDocument();
+});
+
+it('Will hide filter if no proposals', () => {
+  render(renderComponent([]));
+  expect(screen.queryByTestId('proposals-list-filter')).not.toBeInTheDocument();
+});
+
+it('Will show filter if there are proposals', () => {
+  render(renderComponent([enactedProposalClosedLastWeek]));
+  expect(screen.queryByTestId('proposals-list-filter')).toBeInTheDocument();
+});
+
+it('Places proposals correctly in open or closed lists', () => {
+  render(
+    renderComponent([
+      openProposalClosesNextWeek,
+      openProposalClosesNextMonth,
+      enactedProposalClosedLastWeek,
+      rejectedProposalClosedLastMonth,
+    ])
+  );
+  const openProposals = screen.getByTestId('open-proposals');
+  const closedProposals = screen.getByTestId('closed-proposals');
+  expect(
+    openProposals.querySelectorAll("[data-testid='proposals-list-item']").length
```

Comment:
chore (blocking) 
```
const openProposals = within(screen.getByTestId('open-proposals'))
openProposals.getAllByTestId("proposals-list-item").length
```

Reply:
I think a few other areas of the tests could be written like this

Reply:
Oh nice, thanks